tests/unit/core/agents/test_request_logger.py

```
@@ -0,0 +1,294 @@
+            parts=[SimpleNamespace(part_kind="tool-call")],
+            tool_calls=[],
+        )
+        log_history_state([msg], 1, logger)
+        assert "has_tool_calls=True" in calls[0]
+
+
+class TestLogSanitizedHistoryState:
+    def test_debug_mode_false_returns_early(self):
+        calls = []
+        logger = SimpleNamespace(debug=lambda msg: calls.append(msg))
+        log_sanitized_history_state([SimpleNamespace()], False, logger)
```

Ruff (0.14.14)

[error] Boolean positional value in function call

(FBT003)